src/config.rs

```diff
@@ -54,6 +54,10 @@ pub struct Config {
 
     pub outbound_router_max_idle_age: Duration,
 
+    /// The maximum number of queries to the Destination service which may be
+    /// active concurrently.
+    pub max_dst_queries: usize,
```

Comment:
WDYT about naming this `destination_client_concurrency_limit`. I think this leaves less room for interpretation